### PR TITLE
MySQL performance improvements.

### PR DESCRIPTION
**Pull request by prioux**
Changed a little bit how some queries are performed to build the
filter lists in the index table headers. Performance improved.

### CODE REVIEW

File: BrainPortal/app/helpers/scope_helper.rb [hunk 1]
```diff
@@ -681,13 +681,10 @@ def self.model_filter_values(model, attribute, label, association: nil)
     # Fetch the main filter values as an array of arrays:
     # [[value, label, count], [...]]
     model
-      .where("#{attribute} IS NOT NULL")
```
Comment (prioux) [inline BrainPortal/app/helpers/scope_helper.rb]: Surprisingly, this makes a huge difference!

File: BrainPortal/lib/view_scopes.rb [hunk 2]
```diff
@@ -189,7 +189,7 @@ def initialize_copy(other)
     #
     # Note that pagination is not applied by default; specify +paginate+ to
     # paginate +collection+.
-    def apply(collection, on_failure: :ignore, paginate: false)
+    def apply(collection, on_failure: :ignore, paginate: false, sorting: true)
```
Comment (remibernard) [inline BrainPortal/lib/view_scopes.rb]: Please do add some documentation for this new parameter.

File: BrainPortal/app/helpers/scope_helper.rb (hunk 1 above)
Comment (remibernard) [inline BrainPortal/app/helpers/scope_helper.rb]: It does leave the possibility of `NULL`s messing up the counts, however.
It was included in the original filtering system, and included here as a safety measure.

I would guess the production environment doesn't have `NULL`s in filterable columns anyway.

File: BrainPortal/app/helpers/scope_helper.rb [hunk 3]
```diff
@@ -681,13 +681,10 @@ def self.model_filter_values(model, attribute, label, association: nil)
...
       .order(label, attribute)
       .group(attribute, label)
       .raw_rows(attribute, "#{label} AS #{label_alias}", "COUNT(#{attribute})")
       .reject { |r| r.first.blank? }
-      .map(&:to_a)
-      .to_a
```
Comment (remibernard) [inline BrainPortal/app/helpers/scope_helper.rb]: The interface docs for this function clearly indicates that it returns arrays of arrays. These two calls are here to ensure that.
Please do update the docs accordingly.

Comment (prioux) [inline BrainPortal/app/helpers/scope_helper.rb]: @remibernard I'm not sure why these lines were there... raw_rows() already returns an array of array, the reject() doesn't change that, so the last two lines don't do anything. The method still always return the same type as before, an array of arrays...

File: BrainPortal/app/helpers/scope_helper.rb (hunk 1 above)
Comment (prioux) [inline BrainPortal/app/helpers/scope_helper.rb]: And if there were any null, they'd be rejected() a few lines later anyway.

File: BrainPortal/lib/view_scopes.rb (hunk 2 above)
Comment (prioux) [inline BrainPortal/lib/view_scopes.rb]: Done; I will wait until we've finished the other discussion about the to_a() in the other file before I commit.

File: BrainPortal/app/helpers/scope_helper.rb (hunk 3 above)
Comment (remibernard) [inline BrainPortal/app/helpers/scope_helper.rb]: I'm pretty sure I didn't look up `raw_rows`' docs and just assumed Enumerable when I wrote this part. But if `raw_rows` does it, you're right that these two lines serve no purpose.

File: BrainPortal/app/helpers/scope_helper.rb (hunk 1 above)
Comment (remibernard) [inline BrainPortal/app/helpers/scope_helper.rb]: They will still be included in the `COUNT`, however.

Comment (prioux) [inline BrainPortal/app/helpers/scope_helper.rb]: Good point but I'll toleratethat if it happens. :-)

File: BrainPortal/app/helpers/scope_helper.rb (hunk 3 above)
Comment (prioux) [inline BrainPortal/app/helpers/scope_helper.rb]: Well, I'm the one who wrote raw_rows() so it yes I can confirm that's what it does. :-)

**Pull request merged**